packages/compass-web/src/index.spec.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { screen, render, cleanup, waitFor } from '@testing-library/react';
+import type { ConnectionOptions } from 'mongodb-data-service';
+import { expect } from 'chai';
+import { CompassWeb } from './';
+import Sinon from 'sinon';
+import EventEmitter from 'events';
+import ConnectionString from 'mongodb-connection-string-url';
+
+function mockDb(name: string) {
+  return { _id: name, name };
+}
+
+class MockDataService extends EventEmitter {
+  constructor(private connectionOptions: ConnectionOptions) {
+    super();
+  }
+  getConnectionString() {
+    return new ConnectionString(this.connectionOptions.connectionString);
+  }
+  getConnectionOptions() {
+    return { connectionString: 'mongodb://localhost:27017' };
+  }
+  getLastSeenTopology() {
+    return { type: 'Unknown', servers: new Map([]) };
+  }
+  instance() {
+    return Promise.resolve({});
+  }
+  currentOp() {
+    return Promise.resolve({});
+  }
+  top() {
+    return Promise.resolve({});
+  }
+  configuredKMSProviders() {
+    return [];
+  }
+  listDatabases() {
+    return Promise.resolve([mockDb('foo'), mockDb('bar'), mockDb('buz')]);
+  }
+  disconnect() {}
+}
+
+describe('CompassWeb', function () {
+  const mockConnectFn = Sinon.spy(
+    ({ connectionOptions }: { connectionOptions: ConnectionOptions }) => {
+      return Sinon.spy(new MockDataService(connectionOptions));
+    }
+  );
+
+  function renderCompassWeb(
+    props: Partial<React.ComponentProps<typeof CompassWeb>> = {},
+    connectFn = mockConnectFn
+  ) {
+    return render(
+      <CompassWeb
+        connectionString="mongodb://localhost:27017"
+        onActiveWorkspaceTabChange={() => {}}
+        {...props}
+        // @ts-expect-error see component props description
+        __TEST_MONGODB_DATA_SERVICE_CONNECT_FN={connectFn}
+      ></CompassWeb>
+    );
+  }
+
+  afterEach(cleanup);
+
+  it('should render CompassWeb and connect using provided connection string', async function () {
+    renderCompassWeb();
+
+    await waitFor(() => {
+      screen.getByText('Connecting to localhost:27017…');
+    });
+
+    expect(mockConnectFn).to.have.been.calledWithMatch({
+      connectionOptions: { connectionString: 'mongodb://localhost:27017' },
+    });
+
+    // Wait for connection to happen and navigation tree to render
+    await waitFor(() => {
+      screen.getByTestId('compass-web-connected');
+      screen.getByRole('button', { name: 'Databases' });
+      screen.getAllByRole('tree');
```

Comment:
Wanted to write more tests here, but ran into an issue with virtual rendering not rendering in jsdom (and the workaround for this requires passing props directly to the virtualised component, [see here](https://github.com/mongodb-js/compass/blob/f48c76c0ba4b3698461312ba0ebfa6f4b1794a8d/packages/compass-databases-navigation/src/databases-navigation-tree.tsx#L152-L160))

We obviously have all of it tested already, but for the purposes of testing the initial state propagation, we'd probably still want to be able to see the sidebar items in these unit tests. I'll figure out something in a follow-up PRs